platform/core/src/types/HangingProtocol.ts

```diff
@@ -106,35 +130,58 @@ type ViewportOptions = {
   customViewportProps?: Record<string, unknown>;
 };
 
-type DisplaySetOptions = {
+export type DisplaySetOptions = {
   // The id is used to choose which display set selector to apply here
   id: string;
   // An offset to allow display secondary series, for example
   // to display the second matching series (displaySetIndex==1)
   // This cannot easily be done with the matching rules directly.
   displaySetIndex?: number;
+
+  // A key to select a display set UID to reuse from the existing sets.
+  reuseId?: string;
+
   // The options to apply to the display set.
   options?: Record<string, unknown>;
 };
 
-type Viewport = {
+export type Viewport = {
   viewportOptions: ViewportOptions;
   displaySets: DisplaySetOptions[];
+  displaySetsByPosition?: Record<string, DisplaySetOptions[]>;
 };
 
-type ProtocolStage = {
-  id: string;
+/**
+ * Defines which viewports are required - either for activating the stage
+ * at all, or for being a preferred stage (activated by default)
+ */
+export type ViewportsRequired = number;
+
+export type StageEnabled = 'disabled' | 'enabled' | 'passive';
+
+export type ProtocolStage = {
+  id?: string;
   name: string;
   viewportStructure: ViewportStructure;
   viewports: Viewport[];
+  enable?: StageEnabled;
+  // The set of viewports that is required for this stage to be used
+  requiredViewports?: ViewportsRequired;
```

Comment:
`ViewportsRequired` is just a `number`, this doesn't makes sense, either type is wrong or variable name